packages/world/src/systems/MineSystem.sol

```diff
@@ -87,8 +97,8 @@ contract MineSystem is System {
   }
 
   function _mine(EntityId caller, Vec3 coord, uint16 toolSlot, bytes calldata extraData) internal returns (EntityId) {
-    caller.activate();
-    (Vec3 callerCoord,) = caller.requireConnected(coord);
+    EnergyData memory callerEnergy = caller.activate();
```

Comment:
nit: should collapse this to uint128 callerEnergy by adding`.energy` after `.activate()` like in the other systems